src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -136,6 +138,14 @@ public void prepareRun(BatchSinkContext context) throws Exception {
       GCPUtils.createBucket(storage, config.getBucket(), config.getLocation(), cmekKeyName);
     }
     this.outputPath = getOutputDir(context);
+
+    // create asset for lineage
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+      ? ReferenceNames.normalizeFqn(config.getPath())
+      : config.getReferenceName();
+    asset = Asset.builder(referenceName)
+      .setFqn(config.getPath()).setLocation(config.getLocation()).build();
```

Comment:
Shouldn't use this location to construct the asset. This location is only used if the bucket does not exist, and will get ignored if it already exists. You should check existence of the bucket and get the correct location like you do for the GCS source